modules/llama_cpp_plugin/include/compiled_model.hpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef LLAMA_CPP_COMPILED_MODEL_HPP
+#define LLAMA_CPP_COMPILED_MODEL_HPP
+
+#include "openvino/runtime/icompiled_model.hpp"
+#include "openvino/runtime/isync_infer_request.hpp"
+#include "llama.h"
+
+namespace ov {
+    namespace llama_cpp_plugin {
+        class LlamaCppSyncInferRequest;
+        class LlamaCppPlugin;
+        class LlamaCppState;
+        class LlamaCppModel: public ICompiledModel {
+        public:
+            LlamaCppModel(const std::shared_ptr<ov::Model>& model,
+                          const std::shared_ptr<const ov::IPlugin>& plugin,
+                          const ov::SoPtr<ov::IRemoteContext>& context,
+                          const std::shared_ptr<ov::threading::ITaskExecutor>& task_executor
+                          );
+
+            LlamaCppModel(const std::shared_ptr<ov::Model>& ov_model,
+                          std::istream& input_file,
+                          const std::shared_ptr<const IPlugin>& plugin);
```

Comment:
The ctors' definitions had only `THROW_NOT_IMPLEMENTED` inside those, but ok, I removed the ctors and now throw not-implementeds in corresponding `compile_model` overloads (can't remove these since they are pure virtual).